catalog/ui/search-ui/search-endpoint/src/test/java/org/codice/ddf/ui/searchui/query/controller/SearchControllerTest.java

```diff
@@ -442,58 +505,54 @@ private void assertReplies(List<Mutable> replies) {
+        return createFramework(2L, 2, 0);
+    }
+
+    private CatalogFramework createFramework(long totalHits, int... pagingResults) {
 
         CatalogFramework framework = mock(CatalogFramework.class);
-        List<Result> results = new ArrayList<Result>();
-
-        for (int i = 0; i < COUNT; i++) {
-            Result result = mock(Result.class);
-
-            MetacardImpl metacard = new MetacardImpl();
-            metacard.setId("Metacard_" + i);
-            metacard.setTitle("Metacard " + i);
-            metacard.setLocation("POINT(" + i + " " + i + ")");
-            metacard.setType(BasicTypes.BASIC_METACARD);
-            metacard.setCreatedDate(TIMESTAMP);
-            metacard.setEffectiveDate(TIMESTAMP);
-            metacard.setExpirationDate(TIMESTAMP);
-            metacard.setModifiedDate(TIMESTAMP);
-            metacard.setContentTypeName("TEST");
-            metacard.setContentTypeVersion("1.0");
-            metacard.setTargetNamespace(URI.create(getClass().getPackage()
-                    .getName()));
-
-            when(result.getDistanceInMeters()).thenReturn(100.0 * i);
-            when(result.getRelevanceScore()).thenReturn(100.0 * (COUNT - i) / COUNT);
-            when(result.getMetacard()).thenReturn(metacard);
-
-            results.add(result);
-        }
 
-        QueryResponse response = new QueryResponseImpl(mock(QueryRequest.class),
-                new ArrayList<Result>(),
-                COUNT);
-        response.getResults()
-                .addAll(results);
+        List<QueryResponse> responseList = new ArrayList<>();
+        for (int resultsPerPage : pagingResults) {
+            List<Result> results = generateResults(resultsPerPage);
+            QueryResponse response = new QueryResponseImpl(mock(QueryRequest.class),
+                    new ArrayList<>(),
+                    totalHits);
+            response.getResults()
+                    .addAll(results);
+            responseList.add(response);
+        }
 
         try {
-            when(framework.query(any(QueryRequest.class))).thenReturn(response);
-        } catch (UnsupportedQueryException e) {
-            LOGGER.debug("Error querying framework", e);
-        } catch (SourceUnavailableException e) {
-            LOGGER.debug("Error querying framework", e);
-        } catch (FederationException e) {
+            when(framework.query(any(QueryRequest.class))).thenReturn(responseList.get(0),
+                    getNextResponse(responseList));
+        } catch (UnsupportedQueryException | SourceUnavailableException | FederationException e) {
             LOGGER.debug("Error querying framework", e);
         }
         return framework;
     }
 
-    /**
-     * Helper methods for generating exception stacks
-     */
-    private Exception generateExceptionStack(int targetSize) {
-        return generateExceptionStack(targetSize, 0, null);
+    private QueryResponse getNextResponse(List<QueryResponse> responseList) {
```

Comment:
✏️  Might consider changing "get" to "pop" to indicate the side-effect of returning top value as well as removing it. A get is more passive and doesn't remove any elements, which may be what future developers might assume when using this method.